src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java

```diff
@@ -280,55 +280,92 @@ else if (tsDiff >= (-(1L << 30) & 0xFFFFFFFFL))
             {
-                if (snDiff == 2)
+                if (bFirstSeqNum != -1 || aLastSeqNum != -1)
                 {
-                    if (aLastSeqNum == -1)
+                    // XXX(bgrozev): for VPX codecs with PictureID we could find
+                    // the start/end even with diff>2 (if PictureIDDiff == 1)
+
+                    // XXX(gp): we don't have the picture ID in FrameDesc and I
+                    // feel it doesn't belong there. We may need to subclass it
+                    // into VPXFrameDesc and H264FrameDesc and move the
+                    // heuristics logic in there.
+
+                    if (snDiff == 2)
                     {
-                        aLastSeqNum = (aMaxSeqNum + 1) & 0xFFFF;
-                        if (logger.isDebugEnabled())
+                        if (aLastSeqNum == -1)
                         {
-                            logger.debug("Guessed frame end=" + aLastSeqNum);
+                            a.setEnd((aMaxSeqNum + 1) & 0xFFFF);
                         }
-                        a.setEnd(aLastSeqNum);
-                    }
-                    else
-                    {
-                        bFirstSeqNum = (bMinSeqNum - 1) & 0xFFFF;
-                        if (logger.isDebugEnabled())
+                        else
                         {
-                            logger.debug("Guessed frame start=" + bFirstSeqNum);
+                            b.setStart((bMinSeqNum - 1) & 0xFFFF);
                         }
-                        b.setStart(bFirstSeqNum);
+
+                        guessed = true;
                     }
                 }
-                else if (snDiff < 2 || snDiff > (-3 & 0xFFFF))
+                else // (bFirstSeqNum == -1 && aLastSeqNum == -1)
                 {
-                    logger.warn("Frame corruption or packets that are out of " +
-                        "order detected.");
+                    if (snDiff == 3)
+                    {
+                        a.setEnd((aMaxSeqNum + 1) & 0xFFFF);
+                        b.setStart((bMinSeqNum - 1) & 0xFFFF);
+                        guessed = true;
+                    }
                 }
             }
             else
             {
-                if (snDiff == 3)
+                if (bFirstSeqNum != -1 || aLastSeqNum != -1)
                 {
-                    bFirstSeqNum = (bMinSeqNum - 1) & 0xFFFF;
-                    aLastSeqNum = (aMaxSeqNum + 1) & 0xFFFF;
-                    if (logger.isDebugEnabled())
+                    if (snDiff == 1)
                     {
-                        logger.debug(
-                            "Guessed frame start=" + bFirstSeqNum
-                                + ",end=" + aLastSeqNum);
-                    }
+                        if (aLastSeqNum == -1)
+                        {
+                            a.setEnd(aMaxSeqNum & 0xFFFF);
+                        }
+                        else
```

Comment:
I don't think there's a problem here, but it seems strange: If (aLastSeqNum!=-1 &&  bFirstSeqNum!=-1), we update "b" and only "b".